tests/test_smart_completion_multiple_schemata.py

```diff
@@ -1,83 +1,53 @@
-                            'Custom': [
-                                ['func4', [], [], [], '', False, False, False]]
-                         },
-            'datatypes': {
-                            'public':   ['typ1', 'typ2'],
-                            'custom':   ['typ3', 'typ4'],
-                         },
-                'foreignkeys': [
-                    ('public', 'users', 'id', 'custom', 'shipments', 'user_id')
-                ],
-            }
+    'tables': {
+        'public': {
+            'users': ['id', 'email', 'first_name', 'last_name'],
+            'orders': ['id', 'ordered_date', 'status'],
+            'select': ['id', 'insert', 'ABC']
+        },
+        'custom': {
+            'users': ['id', 'phone_number'],
+            'Users': ['userid', 'username'],
+            'products': ['id', 'product_name', 'price'],
+            'shipments': ['id', 'address', 'user_id']
+        },
+        'Custom': {
+            'projects': ['projectid', 'name']
+        }},
+    'functions': {
+        'public': [
+            ['func1', [], [], [], '', False, False, False],
+            ['func2', [], [], [], '', False, False, False]],
+        'custom': [
+            ['func3', [], [], [], '', False, False, False],
+            ['set_returning_func', ['x'], ['integer'], ['o'],
+                'integer', False, False, True]],
+        'Custom': [
+            ['func4', [], [], [], '', False, False, False]]
+     },
+    'datatypes': {
+        'public': ['typ1', 'typ2'],
+        'custom': ['typ3', 'typ4'],
+     },
+    'foreignkeys': {
+        'custom': [
+            ('public', 'users', 'id', 'custom', 'shipments', 'user_id')
+    ]},
+}
+
+testdata = MetaData(metadata)
 
 @pytest.fixture
 def completer():
-
-    import pgcli.pgcompleter as pgcompleter
-    comp = pgcompleter.PGCompleter(smart_completion=True)
-
-    schemata, tables, columns = [], [], []
-
-    for schema, tbls in metadata['tables'].items():
-        schemata.append(schema)
-
-        for table, cols in tbls.items():
-            tables.append((schema, table))
-            # Let all columns be text columns
-            columns.extend([(schema, table, col, 'text') for col in cols])
-
-    functions = [FunctionMetadata(schema, *func_meta)
-                    for schema, funcs in metadata['functions'].items()
-                    for func_meta in funcs]
-
-    datatypes = [(schema, datatype)
-                    for schema, datatypes in metadata['datatypes'].items()
-                    for datatype in datatypes]
-
-    foreignkeys = [ForeignKey(*fk) for fk in metadata['foreignkeys']]
-
-    comp.extend_schemata(schemata)
-    comp.extend_relations(tables, kind='tables')
-    comp.extend_columns(columns, kind='tables')
-    comp.extend_functions(functions)
-    comp.extend_datatypes(datatypes)
-    comp.extend_foreignkeys(foreignkeys)
-    comp.set_search_path(['public'])
-
-    return comp
+    return testdata.completer
```

Comment:
This makes every test share the same completer object. This would be bad if the completer's state changes during a test, because then the tests depend on what order they're run in. I think what we want here is `return testdata.get_completer()` and not have the MetaData object store a reference to the completer